packages/devtools_app/lib/src/shared/directory_picker.dart

```diff
@@ -0,0 +1,84 @@
+// Copyright 2020 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:devtools_app_shared/ui.dart';
+import 'package:flutter/material.dart';
+
+class DirectoryPicker extends StatefulWidget {
+  const DirectoryPicker({
+    required this.onDirectoryPicked,
+    required this.directoryPickerText,
+    this.enabled = true,
+    this.gaSelectionAction,
+    super.key,
+  });
+
+  /// The title of the directory picker button.
+  final String? directoryPickerText;
+
+  final bool enabled;
+
+  final ValueChanged<String> onDirectoryPicked;
+
+  final String? gaSelectionAction;
+
+  @override
+  State<DirectoryPicker> createState() => _DirectoryPickerState();
+}
+
+class _DirectoryPickerState extends State<DirectoryPicker> {
+  final TextEditingController controller = TextEditingController();
+
+  @override
+  void dispose() {
+    controller.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final rowHeight = defaultButtonHeight;
+    return Row(
+      mainAxisAlignment: MainAxisAlignment.center,
+      children: [
+        const Spacer(),
+        Flexible(
+          flex: 4,
+          fit: FlexFit.tight,
+          child: RoundedOutlinedBorder(
+            child: Container(
+              height: rowHeight,
+              padding: const EdgeInsets.symmetric(horizontal: defaultSpacing),
+              child: _buildDirectoryDisplay(),
```

Comment:
since this build method is already small, consider placing the TextField here directly instead of creating a helper for this